napari_animation/utils.py

```diff
@@ -31,3 +31,18 @@ def pairwise(iterable):
     a, b = itertools.tee(iterable)
     next(b, None)
     return zip(a, b)
+
+
+def layer_attribute_changed(value, original_value):
+    """Recursively check if a layer attribute has changed."""
+    if isinstance(value, dict):
+        if (
+            not isinstance(original_value, dict)
+            or value.keys() != original_value.keys()
+        ):
+            return True
+        for key in value.keys():
+            if layer_attribute_changed(value[key], original_value[key]):
+                return True
+    else:
+        return not np.array_equal(value, original_value)
```

Comment:
I goofed here, you're right that the dict was slipping through. However, there's also a case where this returns `None`. This change should take care of it:

```suggestion
def layer_attribute_changed(value, original_value):
    """Recursively check if a layer attribute has changed."""
    if isinstance(value, dict):
        if (
            not isinstance(original_value, dict)
            or value.keys() != original_value.keys()
        ):
            return True
        return any(
            layer_attribute_changed(value[key], original_value[key])
            for key in value.keys()
        )
    return not np.array_equal(value, original_value)
```